services/src/main/java/org/exoplatform/addons/gamification/storage/dao/RuleDAO.java

```diff
@@ -203,6 +200,9 @@ private <T> void addQueryFilterParameters(RuleFilter filter, TypedQuery<T> query
     if (CollectionUtils.isNotEmpty(filter.getSpaceIds())) {
       query.setParameter("ids", filter.getSpaceIds());
     }
+    if (filter.getDateFilterType () != null && ! DateFilterType.ALL.equals(filter.getDateFilterType ())) {
```

Comment:
The operators `==` && `!=` can be used on enums : 

```suggestion
    DateFilterType dateFilterType = filter.getDateFilterType ()
    if (dateFilterType != null && dateFilterType != DateFilterType.ALL) {
```